routes/index.js

```diff
@@ -18,79 +24,186 @@ routes.POSTlogin = function(req, res, next) {
+routes.GETemailver = function(req, res) {
   // User gets here after clicking the link in the email
   // Hence needs to be redirected
   User.verifyEmail(req.query.authToken, function(err, existingAuthToken) {
-    if(err) console.log('err:', err);
+    if (err) console.log('err:', err);
 
     res.redirect('/')
   });
 }
 
-routes.GETindex = function(req, res){
-  // Student.find
-  // res.JSON({students: allStudents});
-  if (req.user)
-    res.sendFile('index.html', { root: path.join(__dirname, '../views') });
-  else
-    res.sendStatus(401);
+routes.GETstudent = function(req, res, next) {
+  Student.findOne({
+    _id: req.params._id
+  }, function(err, currentStudent) {
+    if (err) {
+      res.send(err)
+    }
+    Student.find({}, function(err, allStudents) {
+      if (err) {
+        res.send(err)
+      }
+      res.json({
+        currentStudent: currentStudent,
+        allStudents: allStudents
+      });
+    })
+  })
 }
 
-routes.GETlogin = function(req, res){
-  res.sendFile('login.html', { root: path.join(__dirname, '../views') });
-}
+routes.POSTeditstudent = function(req, res, next) {
 
-routes.GETprogram = function(req, res, next) {
-  res.sendFile('index.html', { root: path.join(__dirname, '../views') });
+  var studentName = req.body.name;
+  var studentProgram = req.body.program;
+  var studentArchived = req.body.archived;
+  var studentID = req.params._id;
+  console.log("student attr", studentName, studentProgram, studentArchived)
+
+  Student.update({
+    _id: studentID
+  }, {
+    name: studentName,
+    program: studentProgram,
+    archived: studentArchived
+  }, function(err, record) {
+    Student.find({}, function(err, allStudents) {
+      if (err) {
+        res.send(err)
+      }
+      res.json(allStudents)
+    })
+  })
 }
 
-routes.GETstudent = function(req, res, next) {
-  res.sendFile('index.html', { root: path.join(__dirname, '../views') });
+routes.POSTaddstudent = function(req, res, next) {
+  var studentName = req.body.name;
+  var studentProgram = req.body.program;
+  var studentArchived = req.body.archived;
+
+  Student.create({
+    name: studentName,
+    program: studentProgram,
+    archived: studentArchived
+  }, function(err, newStudent) {
+    if (err) {
+      res.send(err)
+    }
+    Student.find({}, function(err, allStudents) {
+      res.json({
+        allStudents: allStudents,
+        newStudent: newStudent
+      });
+    })
+  })
 }
 
-routes.POSTeditstudent = function(req, res, next) {
-  res.sendFile('index.html', { root: path.join(__dirname, '../views') });
-}
+routes.POSTnewEntry = function(req, res, next) {
+  // var currentDate = new Date();
+  var studentID = req.params._id;
+  var studentAttendance = req.body.attendance;
+  var studentEntry = req.body.entry;
+  var studentGrades = req.body.grades;
+  var date = req.body.date;
+  console.log("formdata", studentID, studentAttendance, studentGrades,
+    studentEntry, date)
+
+  Attendance.create({
```

Comment:
http://wildermuth.com/2013/8/3/JavaScript_Promises
for async calls, i don't remember it well, but might be useful